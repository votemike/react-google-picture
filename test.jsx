import {render} from '@testing-library/react';
import GooglePicture from "./index";

it('renders a webp source with a 1x and a 2x pixel density, a jpeg source with a 1x and a 2x pixel density and an img inside a picture tag', () => {
  const {container} = render(
    <GooglePicture alt="The alt" imageFormat="jpeg" imageWidth={600} url="https://lh3.googleusercontent.com/ABC123etc"/>
  );
  expect(container).toMatchSnapshot();
});

it('renders a webp source with a 1x and a 2x pixel density, a png source with a 1x and a 2x pixel density and an img inside a picture tag', () => {
  const {container} = render(
    <GooglePicture alt="The alt" imageFormat="png" imageWidth={600} url="https://lh3.googleusercontent.com/ABC123etc"/>
  );
  expect(container).toMatchSnapshot();
});
