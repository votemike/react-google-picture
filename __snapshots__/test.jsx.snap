// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a 480px wide image on screens 600px and smaller and renders a 600px image on any device larger 1`] = `
<div>
  <picture>
    <source
      sizes="(max-width: 600px) 480px, 600px"
      srcset="https://lh3.googleusercontent.com/ABC123etc=w480-rw 480w, https://lh3.googleusercontent.com/ABC123etc=w960-rw 960w, https://lh3.googleusercontent.com/ABC123etc=w600-rw 600w, https://lh3.googleusercontent.com/ABC123etc=w1200-rw 1200w"
      type="image/webp"
    />
    <source
      sizes="(max-width: 600px) 480px, 600px"
      srcset="https://lh3.googleusercontent.com/ABC123etc=w480-rp 480w, https://lh3.googleusercontent.com/ABC123etc=w960-rp 960w, https://lh3.googleusercontent.com/ABC123etc=w600-rp 600w, https://lh3.googleusercontent.com/ABC123etc=w1200-rp 1200w"
      type="image/png"
    />
    <img
      alt="The alt"
      src="https://lh3.googleusercontent.com/ABC123etc=w600-rp"
    />
  </picture>
</div>
`;

exports[`renders a webp source with a 1x and a 2x pixel density, a jpeg source with a 1x and a 2x pixel density and an img inside a picture tag 1`] = `
<div>
  <picture>
    <source
      srcset="https://lh3.googleusercontent.com/ABC123etc=w1200-rw 2x, https://lh3.googleusercontent.com/ABC123etc=w600-rw 1x"
      type="image/webp"
    />
    <source
      srcset="https://lh3.googleusercontent.com/ABC123etc=w1200-rj 2x, https://lh3.googleusercontent.com/ABC123etc=w600-rj 1x"
      type="image/jpeg"
    />
    <img
      alt="The alt"
      src="https://lh3.googleusercontent.com/ABC123etc=w600-rj"
    />
  </picture>
</div>
`;

exports[`renders a webp source with a 1x and a 2x pixel density, a png source with a 1x and a 2x pixel density and an img inside a picture tag 1`] = `
<div>
  <picture>
    <source
      srcset="https://lh3.googleusercontent.com/ABC123etc=w1200-rw 2x, https://lh3.googleusercontent.com/ABC123etc=w600-rw 1x"
      type="image/webp"
    />
    <source
      srcset="https://lh3.googleusercontent.com/ABC123etc=w1200-rp 2x, https://lh3.googleusercontent.com/ABC123etc=w600-rp 1x"
      type="image/png"
    />
    <img
      alt="The alt"
      src="https://lh3.googleusercontent.com/ABC123etc=w600-rp"
    />
  </picture>
</div>
`;
